src/main/java/org/verdictdb/jdbc41/VerdictStreamResultSet.java

```diff
@@ -0,0 +1,1362 @@
+    //    else if (expected.equals("date")) {
+    //      return  actual.equals("timestamp") || actual.equals("date") || actual.equals("time");
+    //    }
+    //    else if (expected.equals("time")) {
+    //      return  actual.equals("timestamp") || actual.equals("date") || actual.equals("time");
+    //    }
+    //    else if (expected.equals("timestamp")) {
+    //      return actual.equals("timestamp") || actual.equals("date") || actual.equals("time");
+    //    }
+    //    else if (expected.equals("asciistream")) {
+    //      return actual.equals("clob");
+    //    }
+    //    else if (expected.equals("binarystream")) {
+    //      return actual.equals("blob") || actual.equals("binary") || actual.equals("varbinary")
+    //          || actual.equals("longvarbinary");
+    //    }
+    //    else if (expected.equals("blob")) {
+    //      return actual.equals("blob");
+    //    }
+    //    else if (expected.equals("clob")) {
+    //      return actual.equals("clob");
+    //    }
+    //    else if (expected.equals("array")) {
+    //      return actual.equals("array");
+    //    }
+    //    else if (expected.equals("ref")) {
+    //      return actual.equals("ref");
+    //    }
+    //    else if (expected.equals("sqlxml")) {
+    //      return actual.equals("xml");
+    //    }
+    //    else if (expected.equals("rowid")) {
+    //      return actual.equals("rowid");
+    //    }
+    //    else if (expected.equals("nclob")) {
+    //      return actual.equals("nclob");
+    //    }
+    //    else return false;
+  }
+
+  @Override
+  public boolean isWrapperFor(Class<?> iface) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public boolean last() throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public void moveToCurrentRow() throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public void moveToInsertRow() throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  /**
+   * Require a mutex here. The stream is executed on the other thread. When the execution of the stream is finished,
+   * it will acquire the mutex. And here are two cases: 1) next() get the mutex, 2) the other thread get the mutex
+   *
+   * 1) If next() first gets the mutex, the flag isCompleted is still false. next() will try to take new queryResult
+   * from queryResults and release the mutex. Then the other thread will take the mutex, append the queryresult and
+   * set the flag to be true.
+   *
+   * 2) If the other thread gets the mutex, it will append the query result and set the flag to be true. Then next()
+   * takes the mutex. Since there are still queryResult in queryResults, next() will take it from the list so that
+   * the last queryResult won't be neglected.
+   *
+   * @return
+   * @throws SQLException
+   */
+  @Override
+  public boolean next() throws SQLException {
```

Comment:
I think `synchronized(this)` keyword could replace this mutex-based mechanism. Please double-check and leave a comment here.